packages/core/contracts/financial-templates/common/financial-product-libraries/LinearContractForDiffrenceFinancialProductLibrary.sol

```diff
@@ -1,43 +1,52 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 pragma solidity ^0.8.0;
-import "./ContractForDifferenceFinancialProductLibrary.sol";
 
+import "@openzeppelin/contracts/utils/math/SignedSafeMath.sol";
+
+import "./ContractForDifferenceFinancialProductLibrary.sol";
 import "../../../common/implementation/Lockable.sol";
 
 contract LinearContractForDiffrenceFinancialProductLibrary is ContractForDifferenceFinancialProductLibrary, Lockable {
     using FixedPoint for FixedPoint.Unsigned;
-    using SafeMath for uint256;
+    using SignedSafeMath for int256;
 
     struct LinearContractForDifferenceParameters {
-        uint256 upperBound;
-        uint256 lowerBound;
+        int256 upperBound;
+        int256 lowerBound;
     }
 
     mapping(address => LinearContractForDifferenceParameters) contractForDifferenceParameters;
 
     function setContractForDifferenceParameters(
         address contractForDifferenceAddress,
-        uint256 upperBound,
-        uint256 lowerBound
+        int256 upperBound,
+        int256 lowerBound
     ) public nonReentrant() {
         require(ExpiringContractInterface(contractForDifferenceAddress).expirationTimestamp() != 0, "Invalid address");
         require(upperBound > lowerBound, "Invalid bounds");
-        require(contractForDifferenceParameters[contractForDifferenceAddress].upperBound > 0, "Parameters already set");
+
+        LinearContractForDifferenceParameters memory params =
+            contractForDifferenceParameters[contractForDifferenceAddress];
+        require(params.upperBound != 0 || params.lowerBound != 0, "Parameters already set");
```

Comment:
hhmmm. this seems redundant. If `upperBound>lowerBound` then surely checking that the `upperBound>0` for the `contractForDifferenceAddress` is sufficient?

Reply:
how you have it is more explicit and does not rely on any dependence between requires but still.

Reply:
It's because they are both integers now. So we assume that at least one of them is nonzero. Technically they could both be negative, or one be zero and the other positive, etc. But it shouldn't be allowed for them both to be zero.